common/djangoapps/nodebb/tasks.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from celery.exceptions import MaxRetriesExceededError
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+
+from common.lib.nodebb_client.client import NodeBBClient
+from django.conf import settings
+
+log = logging.getLogger('edx.celery.task')
+
+
+@task(bind=True, default_retry_delay=settings.NODEBB_TASK_DELAY, max_retries=settings.NODEBB_TASK_MAX_TRIES)
+def create_category_on_nodebb_task(self, course_name):
+    status_code, response_body = NodeBBClient().categories.create(name=course_name)
+
+    if status_code != 200:
+        try:
+            self.retry()
+        except MaxRetriesExceededError:
+            log.error(
```

Comment:
we log message like this to avoid unicode format exceptions where there is unicode data in message .i.e `course_name`

```
log.error("...given course %s due to %s", course_name, response_body)
```
Please change all log statements to use this string formatting.